### PR TITLE
R biodb

### PR DESCRIPTION
**Pull request by pkrog**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

The recipe does not pass the circleci local test, for a reason I fell to understand. See issue #8649 .

### CODE REVIEW

**Comment by pkrog** [comment]
Please @bioconda/r would you have any advice or idea about the failure met while testing locally with circleci (see issue #8649 ).

**Comment by pkrog** [comment]
I've tested bioconda/circleci with a very simple R package with C++ code (one simple basic function) using Rcpp that I've built myself. The package had no other dependences. CircleCI gave me the same error exactly.

**Comment by cbrueffer** [comment]
Have you tried adding ```libgfortran``` as dependency? See ```recipes/bioconductor-fabia``` for an R example that does the same.

File: recipes/r-biodb/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#EOF
+echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+cat /proc/version
+echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+cat /etc/issue
+echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+env
+echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+alias
+echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Comment (epruesse) [inline recipes/r-biodb/build.sh]: BTW: I used to just put a line `sh` in here (or `bash --norc`) to get "into" the environment and be able to poke around. Not sure if that still works with the docker - it does with `conda build` though.

**Comment by epruesse** [comment]
It's definitely just the linker not finding libgfortran - header files are there though.

How does Rcpp decide which libs it needs? Is there a run dependency missing on one of the included packages?

**Comment by pkrog** [comment]
It's not linked to other packages, since I've tried with a simple recipe (not dependencies except Rcpp) and I get the same error.
What is weird, is that when I compile my package under macOS, Ubuntu or CentOS, the -lgfortran flag is not used.

**Comment by pkrog** [comment]
For instance if I make a docker like this for compiling my package under CentOS:
```
FROM centos:latest

MAINTAINER Pierrick Roger ( pk.roger@icloud.com )

RUN curl -O https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
RUN yum install -y bzip2
RUN bash Miniconda3-latest-Linux-x86_64.sh -b -p "/miniconda3"
ENV PATH="/miniconda3/bin:$PATH"

RUN conda create -n biodb r-rcpp
RUN bash -c ". activate biodb && conda install r-xml r-stringr r-plyr r-jsonlite r-bitops r-rcurl r-r.utils r-digest r-codetools"
RUN yum install -y make
RUN bash -c ". activate biodb && gcc --version"

ADD . /biodb
RUN bash -c ". activate biodb && R CMD INSTALL --build /biodb"
```
Everything works fine and the compiling lines are:
```
x86_64-conda_cos6-linux-gnu-cc -I/miniconda3/envs/biodb/lib/R/include -DNDEBUG  -I"/miniconda3/envs/biodb/lib/R/library/Rcpp/include" -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -I/miniconda3/envs/biodb/include   -fpic  -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -pipe -I/miniconda3/envs/biodb/include -fdebug-prefix-map==/usr/local/src/conda/- -fdebug-prefix-map==/usr/local/src/conda-prefix  -c closeMatchPpm.c -o closeMatchPpm.o
x86_64-conda_cos6-linux-gnu-cc -I/miniconda3/envs/biodb/lib/R/include -DNDEBUG  -I"/miniconda3/envs/biodb/lib/R/library/Rcpp/include" -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -I/miniconda3/envs/biodb/include   -fpic  -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -pipe -I/miniconda3/envs/biodb/include -fdebug-prefix-map==/usr/local/src/conda/- -fdebug-prefix-map==/usr/local/src/conda-prefix  -c registerDynamicSymbol.c -o registerDynamicSymbol.o
x86_64-conda_cos6-linux-gnu-cc -shared -L/miniconda3/envs/biodb/lib/R/lib -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-rpath,/miniconda3/envs/biodb/lib -L/miniconda3/envs/biodb/lib -o biodb.so closeMatchPpm.o registerDynamicSymbol.o -L/miniconda3/envs/biodb/lib/R/lib -lR
```

**Comment by pkrog** [comment]
Hi @cbrueffer, yes I've tried with `libgfortran`, no difference.

**Comment by pkrog** [comment]
Thanks @cbrueffer , your recipe did help me well. I've added gcc as a build requirement, so now it uses this gcc instead of the one supplied by R, and it works.

**Comment by pkrog** [comment]
I fail to understand. `circleci build` works now perfectly on my local computer. Why is not building the shared library `biodb.so` on the circleci server?

**Comment by pkrog** [comment]
Is there a way to run this "mulled test" locally?

**Comment by daler** [comment]
@pkrog you can use the second method described here (the bootstrap method): https://bioconda.github.io/contribute-a-recipe.html#test-locally

**Pull request merged**